src/components/app/routes/data/utils.js

```diff
@@ -269,50 +262,103 @@ export async function ensureAuthenticatedUser(requestUrl, params) {
+      },
+    );
+    if (enterpriseSlug && foundEnterpriseCustomerUserForSlug) {
+      nextActiveEnterpriseCustomer = foundEnterpriseCustomerUserForSlug.enterpriseCustomer;
+    }
   }
 
-  // Otherwise, try to find the enterprise customer for the given slug and, if found, update it
-  // as the active enterprise customer for the learner.
-  const foundEnterpriseCustomerUserForSlug = allLinkedEnterpriseCustomerUsers.find(
-    enterpriseCustomerUser => {
-      if (!enterpriseCustomerUser.enterpriseCustomer) {
-        return false;
-      }
-      return enterpriseCustomerUser.enterpriseCustomer.slug === enterpriseSlug;
-    },
-  );
-  if (enterpriseSlug && foundEnterpriseCustomerUserForSlug) {
-    const {
-      enterpriseCustomer: nextActiveEnterpriseCustomer,
-    } = foundEnterpriseCustomerUserForSlug;
-    // Makes the POST API request to update the active enterprise customer
-    // for the learner in the backend for future sessions.
-    await updateUserActiveEnterprise({ enterpriseCustomer: nextActiveEnterpriseCustomer });
+  // If we've determined that the active enterprise customer user should be updated, update it.
+  if (nextActiveEnterpriseCustomer) {
+    try {
+      await updateUserActiveEnterprise({ enterpriseCustomer: nextActiveEnterpriseCustomer });
+    } catch (error) {
+      logError(`Unable to update active enterprise customer: ${nextActiveEnterpriseCustomer}
+      for user ${authenticatedUser.userId}
+      ${error.message}`);
+      return {
+        enterpriseCustomer,
+        allLinkedEnterpriseCustomerUsers,
+      };
+    }
+    // If the active enterprise customer user was updated, override the previous active
+    // enterprise customer user data with the new active enterprise customer user data
+    // for subsequent queries.
     const updatedLinkedEnterpriseCustomerUsers = allLinkedEnterpriseCustomerUsers.map(
       ecu => ({
         ...ecu,
-        active: !!(ecu.enterpriseCustomer?.uuid === nextActiveEnterpriseCustomer.uuid),
+        active: ecu.enterpriseCustomer?.uuid === nextActiveEnterpriseCustomer.uuid,
       }),
     );
+    enterpriseCustomer = nextActiveEnterpriseCustomer;
+    allLinkedEnterpriseCustomerUsers = updatedLinkedEnterpriseCustomerUsers;
+    // Optimistically update the BFF layer (use helper)
+    if (matchedBFFQuery) {
+      queryClient.setQueryData(matchedBFFQuery({ enterpriseSlug }), {
+        ...queryClient.getQueryData(matchedBFFQuery({ enterpriseSlug })),
+        enterpriseCustomer,
+        activeEnterpriseCustomer,
+        allLinkedEnterpriseCustomerUsers: updatedLinkedEnterpriseCustomerUsers,
+      });
+    }
     return {
-      enterpriseCustomer: nextActiveEnterpriseCustomer,
-      updatedLinkedEnterpriseCustomerUsers,
+      enterpriseCustomer,
+      allLinkedEnterpriseCustomerUsers,
     };
   }
-  if (staffEnterpriseCustomer) {
-    return null;
+
+  // If the active enterprise customer user was not updated or enterpriseSlug is missing, return null.
```

Comment:
Should this comment be updated to reflect the below code? E.g.

```suggestion
  // Given the user has an active ECU, but the current route has no slug, redirect to the slug of the active ECU.
```